Prisma/Chat/ChatView.swift

```diff
@@ -36,6 +63,53 @@ struct ChatView: View {
     }
 }
 
+extension ChatView {
+    func signInWithFirebase() async throws {
+        token = try await Auth.auth().currentUser?.getIDToken()
+    }
+    
+    func generateJWT(completion: @escaping (String?) -> Void ) {
+        if let currentUser = Auth.auth().currentUser {
+            // Generating JWT Token
+            currentUser.getIDTokenForcingRefresh(true) { (token, error) in
+                if let error = error {
+                    print("Error getting ID token: \(error.localizedDescription)")
+                    completion(nil)
+                } else if let token = token { // Setting the JWT token and send it to chat
+                    print("JWT is: \(token)")
+                    sendTokenToBackend(token: token, completion: completion)
+                } else {
+                    print("No token received.")
+                    completion(nil)
+                }
+            }
+        } else {
+            print("No current user")
+            completion(nil)
+        }
+    }
+    
+    func sendTokenToBackend(token: String, completion: @escaping (String?) -> Void) {
```

Comment:
Why is this function needed? I thought the frontend would take the token from the URL parameter and use that to authenticate the frontend <-> backend connection.